### PR TITLE
AuthZ: Implementation, tests and docs

### PR DESCRIPTION
**Pull request by pangea-andrest**

### CODE REVIEW

File: pangea-sdk/v3/service/authz/api.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+	Namespace string  `json:"namespace"`
+	Action    string  `json:"action"`
+	Subject   Subject `json:"subject"`
+}
+
+type ListResourcesResult struct {
+	IDs []string `json:"ids"`
+}
+
+// @summary List resouces
```
Comment (cj-f) [inline pangea-sdk/v3/service/authz/api.go:269]: List resources

File: pangea-sdk/v3/service/authz/api.go [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+//
+// @description Given a namespace, action, and subject, list all the resources in the namespace that the subject has the action with.
```
Comment (cj-f) [inline pangea-sdk/v3/service/authz/api.go:271]: has permission to the action with.

**Review by kenany — APPROVED** [review]

**Comment by pangea-andrest** [comment]
Rebase on main and change target to `beta`

File: pangea-sdk/v3/service/authz/api.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+	Tuples []Tuple `json:"tuples"`
+}
+
+type TupleCreateResult struct {
+}
+
+// @summary Write tuples.
+//
+// @description Write tuples. The request will fail if there is no schema or the tuples do not validate against the schema.
```
Comment (cj-f) [inline pangea-sdk/v3/service/authz/api.go:38]: Write tuples. The request will fail if tuples do not validate against the schema defined resource types.

I'm updating the API description as well

**Review by kenany — APPROVED** [review]

**Pull request merged**